src/llscan.cc

```diff
@@ -0,0 +1,376 @@
+      v8::Error err;
+      v8::Value v8_value(&llv8, *it);
+      std::string res = v8_value.Inspect(&inspect_options, err);
+      result.Printf("%s\n", res.c_str());
+    }
+
+  } else {
+    result.Printf("No objects found with type name %s\n", type_name.c_str());
+  }
+
+  return true;
+}
+
+
+FindJSObjectsVisitor::FindJSObjectsVisitor(SBTarget& target,
+                                           TypeRecordMap& mapstoinstances)
+    : target_(target), mapstoinstances_(mapstoinstances) {
+  found_count_ = 0;
+  address_byte_size_ = target_.GetProcess().GetAddressByteSize();
+  // Load V8 constants from postmortem data
+  llv8.Load(target);
+}
+
+
+/* Visit every address, a bit brute force but it works. */
+uint64_t FindJSObjectsVisitor::Visit(uint64_t location, uint64_t available) {
+  lldb::SBError error;
+
+  // Test if the map points to a real map.
+  // Try to create an object out of it.
+
+  uint64_t word = target_.GetProcess().ReadUnsignedFromMemory(
+      location, target_.GetProcess().GetAddressByteSize(), error);
+
+  v8::Value v8_value(&llv8, word);
+
+  v8::Error err;
+  // Test if this is SMI
+  // Skip inspecting things that look like Smi's, they aren't objects.
+  v8::Smi smi(v8_value);
+  if (smi.Check()) {
+    return address_byte_size_;
+  }
+
+  v8::HeapObject heap_object(v8_value);
+  if (!heap_object.Check()) {
+    return address_byte_size_;
+  }
+  if (heap_object.IsHoleOrUndefined(err)) {
+    return address_byte_size_;
+  }
+  if (err.Fail()) {
+    return address_byte_size_;
+  }
+
+  v8::HeapObject map_object = heap_object.GetMap(err);
+  if (err.Fail() || !map_object.Check()) {
+    return address_byte_size_;
+  }
+
+  if (!IsAHistogramType(heap_object, err)) {
+    return address_byte_size_;
+  }
+
+  if (err.Fail()) {
+    return address_byte_size_;
+  }
+
+  v8::Map map(map_object);
+
+  v8::Value::InspectOptions inspect_options;
+  inspect_options.detailed = false;
+  inspect_options.print_map = false;
+  inspect_options.string_length = 0;
+
+  std::string type_name = heap_object.GetTypeName(&inspect_options, err);
+
+  /* No entry in the map, create a new one. */
+  if (mapstoinstances_.count(type_name) == 0) {
+    TypeRecord* t = new TypeRecord();
+    t->type_name = new std::string(type_name);
+    t->instances = new std::set<uint64_t>();
+    t->instances->insert(word);
+    t->instance_count = 1;
+    t->total_instance_size = map.InstanceSize(err);
+
+    mapstoinstances_.insert(std::pair<std::string, TypeRecord*>(type_name, t));
+  } else {
+    /* Update an existing instance, if we haven't seen this instance before. */
+    TypeRecord* t = mapstoinstances_.at(type_name);
+    /* Determine if this is a new instance.
+     * (We are scanning pointers to objects, we may have seen this location
+     * before.)
+     */
+    if (t->instances->count(word) == 0) {
+      t->instances->insert(word);
+      t->instance_count++;
+      t->total_instance_size += map.InstanceSize(err);
```

Comment:
Once these will become private members, we will need to have a method to do this accounting.